### ISSUE TITLE
Unable to push to ECR private repo in 1.0.7

### ISSUE THREAD

**Issue by gdrapp** [hashicorp/packer-plugin-docker#108]
#### Overview of the Issue

1.0.7 introduces some new code for ECR repos but it appears to fail to parse an ECR private repo login_server correctly. Reverting to 1.0.6 fixes the issue.

#### Reproduction Steps

packer build

### Plugin and Packer version

Packer version 1.8.0
Docker plugin 1.0.7

### Simplified Packer Buildfile

```hcl
post-processor "docker-push" {
      ecr_login    = true
      login_server = "317215053986.dkr.ecr.us-east-2.amazonaws.com"
    }
```

### Operating system and Environment details

macOS 12.4, M1 processor

### Log Fragments and crash.log files
```bash
* Post-processor failed: failed to parse the ECR URL: 317215053986.dkr.ecr.us-east-2.amazonaws.com
parse "317215053986.dkr.ecr.us-east-2.amazonaws.com": invalid URI for request
it should be either of the form `public.ecr.aws/<registry_alias>/<registry_name>` or `<account number>.dkr.ecr.<region>.amazonaws.com`
```

**Comment by devashish-patel** [comment]
Hi @gdrapp,

In the given snippet, the `login_server` argument is missing `https://` and `repository` name.  Please add `https://` and the `repo` name to the `login_server` argument in your `post-processor` and everything should work fine.

Example:
```hcl
    post-processor "docker-push" {
      ecr_login    = true
      aws_profile  = "default"
      login_server = "https://317215053986.dkr.ecr.us-east-2.amazonaws.com/<YOUR_PRIVATE_REPO_NAME>"
    }
```

Full example template:
```hcl
packer {
  required_plugins {
    docker = {
      version = ">= 1.0.7"
      source  = "github.com/hashicorp/docker"
    }
  }
}

source "docker" "ubuntu-bionic" {
  image  = "ubuntu:bionic"
  commit = true
}

build {
  name = "docker-push-ecr-private"
  sources = [
    "source.docker.ubuntu-bionic"
  ]

  post-processors {
    post-processor "docker-tag" {
      repository = "746700064644.dkr.ecr.us-east-1.amazonaws.com/private_dev_hc_pkr_dkr_test_1"
      tags       = ["latest"]
    }

    post-processor "docker-push" {
      ecr_login    = true
      aws_profile  = "default"
      login_server = "https://746700064644.dkr.ecr.us-east-1.amazonaws.com/private_dev_hc_pkr_dkr_test_1"
    }
  }
}
```

This should fix the problem otherwise feel free to reach out!

Thanks!

**Comment by Tocyuki** [comment]
@devashish-patel @gdrapp 
I'm having a similar issue in my environment.
Provisioning succeeded, only docker push failed.
In my case,  reverting to 1.0.6 not fixes the issue.

## Reproduction Steps
packer build

## Plugin and Packer version
- Packer version 1.8.2
- Docker plugin 1.0.7

## Operating system and Environment details
- macOS 12.5, M1 processor

## Debug Log
```
2022/08/03 11:16:54 packer-plugin-docker_v1.0.7_x5.0_darwin_arm64 plugin: 2022/08/03 11:16:54 Executing: /opt/homebrew/bin/docker [--config /var/folders/l3/cjhrnn8x3cd7jwsh6wf5zwgm0000gq/T/packer2063444993 logout https://xxxxxxxxxxxx.dkr.ecr.ap-northeast-1.amazonaws.com/dev-app]
    docker.app (docker-push): Removing login credentials for xxxxxxxxxxxx.dkr.ecr.ap-northeast-1.amazonaws.com
    docker.app (docker-push): Removing temporary Docker configuration directory
2022/08/03 11:16:54 packer-plugin-docker_v1.0.7_x5.0_darwin_arm64 plugin: 2022/08/03 11:16:54 error: Bad exit status: 1
2022/08/03 11:16:54 [INFO] (telemetry) ending docker-push

* Post-processor failed: Bad exit status: 1
Build 'docker.app' errored after 1 minute 53 seconds: 1 error(s) occurred:

* Post-processor failed: Bad exit status: 1
```

## Example Packer template
```hcl
variable "env" {}
variable "service" {}

variable "aws_access_key" {
  type      = string
  default   = "${env("AWS_ACCESS_KEY_ID")}"
  sensitive = true
}

variable "aws_secret_key" {
  type      = string
  default   = "${env("AWS_SECRET_ACCESS_KEY")}"
  sensitive = true
}

variable "aws_account_id" {
  type    = string
  default = "xxxxxxxxxxxx"
}

variable "region" {
  type    = string
  default = "ap-northeast-1"
}

variable "name" {
  type    = string
  default = "my-app"
}

packer {
  required_plugins {
    docker = {
      version = "1.0.7"
      source  = "github.com/hashicorp/docker"
    }
  }
}

source "docker" "app" {
  image          = "amazonlinux:2"
  commit         = true
  platform       = "linux/x86_64"
}

build {
  sources = [
    "source.docker.app"
  ]

  provisioner "shell" {
    inline = [
      "yum install -y sudo python3"
    ]
  }

  provisioner "ansible" {
    playbook_file       = "${var.service}.yml"
    inventory_directory = "inventories/${var.env}"
    groups              = [var.service]
    extra_arguments = [
      "--extra-vars",
      "ansible_python_interpreter=/usr/bin/python3"
    ]
  }

  post-processors {
    post-processor "docker-tag" {
      repository = "${var.aws_account_id}.dkr.ecr.${var.region}.amazonaws.com/dev-app"
      tags       = ["latest"]
    }

    post-processor "docker-push" {
      ecr_login      = true
      aws_access_key = var.aws_access_key
      aws_secret_key = var.aws_secret_key
      login_server   = "https://${var.aws_account_id}.dkr.ecr.${var.region}.amazonaws.com/dev-app"
    }
  }
}
```

**Comment by axway-dpal** [comment]
> Hi @gdrapp,
> 
> In the given snippet, the `login_server` argument is missing `https://` and `repository` name. Please add `https://` and the `repo` name to the `login_server` argument in your `post-processor` and everything should work fine.
> 
> Example:
> 
> ```terraform
>     post-processor "docker-push" {
>       ecr_login    = true
>       aws_profile  = "default"
>       login_server = "https://317215053986.dkr.ecr.us-east-2.amazonaws.com/<YOUR_PRIVATE_REPO_NAME>"
>     }
> ```
> 
> Full example template:
> 
> ```terraform
> packer {
>   required_plugins {
>     docker = {
>       version = ">= 1.0.7"
>       source  = "github.com/hashicorp/docker"
>     }
>   }
> }
> 
> source "docker" "ubuntu-bionic" {
>   image  = "ubuntu:bionic"
>   commit = true
> }
> 
> build {
>   name = "docker-push-ecr-private"
>   sources = [
>     "source.docker.ubuntu-bionic"
>   ]
> 
>   post-processors {
>     post-processor "docker-tag" {
>       repository = "746700064644.dkr.ecr.us-east-1.amazonaws.com/private_dev_hc_pkr_dkr_test_1"
>       tags       = ["latest"]
>     }
> 
>     post-processor "docker-push" {
>       ecr_login    = true
>       aws_profile  = "default"
>       login_server = "https://746700064644.dkr.ecr.us-east-1.amazonaws.com/private_dev_hc_pkr_dkr_test_1"
>     }
>   }
> }
> ```
> 
> This should fix the problem otherwise feel free to reach out!
> 
> Thanks!

isn't this is a breaking change?

**Comment by axway-dpal** [comment]
Another Interesting case for me.
When I use private repo from `us-east-1` like `xxxxxxxxxxxx.dkr.ecr.us-east-1.amazonaws.com`
Sample Source Code:
``` hcl
variable "registry" {
  type    = string
  default = "xxxxxxxxxxxx.dkr.ecr.us-east-1.amazonaws.com"
}
source "docker" "base" {
  image  = "python:3.9-alpine"
  commit = true
  changes = [
    "ENTRYPOINT /bin/bash",
    "ENV BASH_ENV /home/gitlab/.bashrc",
    "USER gitlab"
  ]
  run_command = var.run_cmd
}

build {

  sources = ["source.docker.base"]

  provisioner "ansible" {
    playbook_file = "prepare.yml"
    user          = "root"
  }

  provisioner "ansible" {
    playbook_file        = "main.yml"
    user                 = "gitlab"
    galaxy_file          = "requirements.yml"
    galaxy_force_install = true
    extra_arguments      = ["--extra-vars", "target=all"]
  }

  post-processors {
    post-processor "docker-tag" {
      repository = local.repo
      tags       = [formatdate("DDMMMYYYYhhmm", timestamp()), "latest"]
    }

    post-processor "docker-push" {
      ecr_login    = true
      login_server = var.registry
    }
  }
}

```
**Docker push works fine**


But when I use another region like `eu-west-3` like  `xxxxxxxx.dkr.ecr.eu-west-3.amazonaws.com` docker push fails with same error.

**Comment by devashish-patel** [comment]
Hi, @axway-dpal and @gdrapp. We have fixed the bug related to this in `v1.0.8`. Please update the plugin and everything should work fine as it used to be (aka no need to prefix the `login_server` value with `https://`! 

Please feel free to reopen this issue, if you have questions or facing any issues.

Thanks for the patience!

**Comment by axway-dpal** [comment]
Thanks a lot @devashish-patel, I will try it asap.
